pkg/backend/openshiftcluster.go

```diff
@@ -361,3 +365,10 @@ func (ocb *openShiftClusterBackend) emitMetrics(doc *api.OpenShiftClusterDocumen
 		"newProvisioningState": string(provisioningState),
 	})
 }
+
+func (ocb *openShiftClusterBackend) setNoPucmPending(ctx context.Context, doc *api.OpenShiftClusterDocument) (*api.OpenShiftClusterDocument, error) {
+	return ocb.dbOpenShiftClusters.Patch(ctx, doc.Key, func(doc *api.OpenShiftClusterDocument) error {
+		doc.OpenShiftCluster.Properties.PucmPending = false
+		return nil
+	})
+}
```

Comment:
In general, I'm not a fan of flag arguments, especially boolean ones. Here is a good justification why: https://martinfowler.com/bliki/FlagArgument.html

Reply:
Are you good if we keep this as is?

Reply:
Hi @niontive lets keep it like it is. Should we ever consider that change, we can always do it.

Reply:
I also like and read Fowler's blog post. 😃  

Just a subtle correction. This is not a [FlagArgument](https://martinfowler.com/bliki/FlagArgument.html) as a FlagArgument is a parameter that it is used inside a function to perform one flow path or another depending on its value (if/else). This is just a setter method on a property of the api.OpenShiftClusterDocument. I think we are more on the _Boolean Setting Method_ section rather than in the _BooleanFlag_ (as we are not talking about how to use the flag, just how to set it). 

I think what you are trying to do here is to create a method that provides a better semantic meaning to the method name (like _setOn_ in the blogpost) rather than doing it explicitly assigning the value from the outside (_setSwitch(boolean on)_ in the blog) but I think you are not doing it like in the blog. 

IMHO if you want to use the technique of the post you should not name the method _setPucmPending_ but rather something like **pucmIsPending** because with the actual name there is no clue from the caller if you are setting it to false or true (just looking into the implementation) as the function name does not give any clue about the value that it is set (it does not provides any semantic meaning). In other words, having the method name as "set" + "PucmPending" (name of the attribute)  does not reflect explicitly this is setting it to true (as you can also say "set PucPending to false"). 

I think what you are trying to achieve is something like:
In the post:                                           in our code it would be:
void setSwitch(boolean on);      ->      setPucmPending(boolean)         // -> you say you don't want this

In the post:                           in our code it would be                                  // I understand this is what you want to achieve
 void setOn();          ->             pucmIspending()
 void setOff();         ->             pucmIspending()

I am good with keeping as it is but if what we want is to use the technique of the blog, I would use a name with a better semantic meaning. I would LGTM anyway as it is just a minor detail 😄